added specific exceptions for http errors 400, 401, 403, 500

JeffreyThiessen opened the pull request:
## Description of changes
Went making requests to IRIDA we now explicitly check for http errors 400, 401, 403 and 500. These are the most common errors that could occur and should be handled separately.

## Related issue
Fixes #31 

## Checklist
Things for the developer to confirm they've done before the PR should be accepted:

* [x] CHANGELOG.md updated with information for new change.
* [x] Tests added (or description of how to test) for any new features.
* [ ] User documentation updated for UI or technical changes.

tom114 commented:
Is this the type of error I'm supposed to see with this?  This was uploading to a project my user didn't have access to:
```
2019-07-18 15:35:01 ERROR    Error while uploading [01-1111]: [Forbidden]
2019-07-18 15:35:01 ERROR    Failed to upload SequencingRun, Could not access resources on IRIDA
Traceback (most recent call last):
  File "upload_run.py", line 199, in <module>
    main()
  File "upload_run.py", line 174, in main
    return upload(args.directory, args.force)
  File "upload_run.py", line 184, in upload
    return core.cli_entry.upload_run_single_entry(run_directory, force_upload)
  File "/.../irida/irida-uploader/core/cli_entry.py", line 44, in upload_run_single_entry
    return _validate_and_upload(directory_status)
  File "/.../irida/irida-uploader/core/cli_entry.py", line 179, in _validate_and_upload
    run_id = api_handler.upload_sequencing_run(sequencing_run)
  File "/.../irida/irida-uploader/core/api_handler.py", line 168, in upload_sequencing_run
    raise e
  File "/.../irida/irida-uploader/core/api_handler.py", line 156, in upload_sequencing_run
    upload_id=run_id)
  File "/.../irida/irida-uploader/api/api_calls.py", line 678, in send_sequence_files
    raise self._get_irida_exception(response)
api.exceptions.irida_resource_error.IridaResourceError: Request to IRIDA is Forbidden. Do you have access to this resource?: 403: Forbidden
```

JeffreyThiessen commented:
> Is this the type of error I'm supposed to see with this? This was uploading to a project my user didn't have access to:

Fixed 2 cases where exceptions were not being handled correctly and added test cases to the unittests suite for them.

tom114 commented:
My testing works out now.  Thanks!  Merging.
